src/engines/preprocessing_engine.py

```diff
@@ -21,22 +28,37 @@ def get_columns(self):
             return []
         return self.state.imported_data.columns
 
+    def get_numeric_columns(self):
+        if self.state.imported_data is None:
+            return []
+        return self.state.imported_data.select_dtypes(include=["number"]).columns
+
+    def get_size(self):
+        if self.state.imported_data is None:
+            return []
+        return len(self.state.imported_data.select_dtypes(include=["number"]))
+
     def set_state(self, columns):
         self.state.imported_data = self.state.raw_data[columns].copy()
 
-    def create_plot(self, column_name, plot_type):
+    def create_plot(self, column_name, plot_type, scatter_settings):
```

Comment:
I think that this method should be in `preprocessing_widget` instead of engine cause it renders GUI